Tools/Algorithms/stencil.py

```diff
@@ -142,87 +144,115 @@ def compute_stencils(coeff_nodal, coeff_stagg, axis):
-          stencil_avg_stagg /= (stencil_stagg.shape[2]*stencil_stagg.shape[0])
-    elif axis == 2:
-          # Averaged over kx and ky
-          stencil_avg_nodal  = np.sum(np.sum(stencil_nodal, axis=1), axis=0)
-          stencil_avg_nodal /= (stencil_nodal.shape[1]*stencil_nodal.shape[0])
-          stencil_avg_stagg  = np.sum(np.sum(stencil_stagg, axis=1), axis=0)
-          stencil_avg_stagg /= (stencil_stagg.shape[1]*stencil_stagg.shape[0])
+    if dims == 1:
+        stencil_avg_nodal = stencil_nodal
+        stencil_avg_stagg = stencil_stagg
+    elif dims == 2:
+        if axis == 0:
+              # Averaged over kz
+              stencil_avg_nodal  = np.sum(stencil_nodal, axis=-1)
+              stencil_avg_nodal /= stencil_nodal.shape[-1]
+              stencil_avg_stagg  = np.sum(stencil_stagg, axis=-1)
+              stencil_avg_stagg /= stencil_stagg.shape[-1]
+        elif axis == 1 or axis == -1:
+              # Averaged over kx
+              stencil_avg_nodal  = np.sum(stencil_nodal, axis=0)
+              stencil_avg_nodal /= stencil_nodal.shape[0]
+              stencil_avg_stagg  = np.sum(stencil_stagg, axis=0)
+              stencil_avg_stagg /= stencil_stagg.shape[0]
+    elif dims == 3:
+        if axis == 0:
+              # Averaged over ky and kz
+              stencil_avg_nodal  = np.sum(np.sum(stencil_nodal, axis=-1), axis=1)
+              stencil_avg_nodal /= (stencil_nodal.shape[-1]*stencil_nodal.shape[1])
+              stencil_avg_stagg  = np.sum(np.sum(stencil_stagg, axis=-1), axis=1)
+              stencil_avg_stagg /= (stencil_stagg.shape[-1]*stencil_stagg.shape[1])
+        elif axis == 1:
+              # Averaged over kx and kz
+              stencil_avg_nodal  = np.sum(np.sum(stencil_nodal, axis=-1), axis=0)
+              stencil_avg_nodal /= (stencil_nodal.shape[-1]*stencil_nodal.shape[0])
+              stencil_avg_stagg  = np.sum(np.sum(stencil_stagg, axis=-1), axis=0)
+              stencil_avg_stagg /= (stencil_stagg.shape[-1]*stencil_stagg.shape[0])
+        elif axis == 2 or axis == -1:
+              # Averaged over kx and ky
+              stencil_avg_nodal  = np.sum(np.sum(stencil_nodal, axis=1), axis=0)
+              stencil_avg_nodal /= (stencil_nodal.shape[1]*stencil_nodal.shape[0])
+              stencil_avg_stagg  = np.sum(np.sum(stencil_stagg, axis=1), axis=0)
+              stencil_avg_stagg /= (stencil_stagg.shape[1]*stencil_stagg.shape[0])
 
     stencils = dict()
     stencils['nodal'] = abs(stencil_avg_nodal)
     stencils['stagg'] = abs(stencil_avg_stagg)
 
     return stencils
 
-def compute_all(dx, dy, dz, dt, nox, noy, noz, v_gal, nx=256, ny=256, nz=256):
+def compute_all(dims, dx_boosted, dt, psatd_order, v_gal, nx=None):
     """
     Compute nodal and staggered stencils along all directions.
 
     Parameters
     ----------
-    dx : float
-        Cell size along x.
-    dy : float
-        Cell size along y.
-    dz : float
-        Cell size along z.
+    dims : int
```

Comment:
Can you get `dims = len(dx_boosted)` instead of requiring it as an input parameter?